build/Microsoft.DotNet.Cli.Prepare.targets

```diff
@@ -53,11 +48,14 @@
       <SharedFrameworkPublishDirectory>$(IntermediateDirectory)/sharedFrameworkPublish</SharedFrameworkPublishDirectory>
     </PropertyGroup>
 
+    <GetCommitHash RepoRoot="$(RepoRoot)">
```

Comment:
I went back & forth on this quite a bit. We need to use ToolTasks to take advantage of MSBuild's command composition/execution layer. We cannot have two commands executed from one ToolTask. CommitCount is tightly coupled to our versioning story, while CommitHash is used in telemetry and accounting. Taking this approach means that we don't need to pass the CommitCount to the version generator, which seems nice at the moment. 

I have another PR baking that further decomposes our version logic. Once that is further along it may make more sense to break apart CommitCount from Version.